cmake/Modules/FindQhull.cmake

```diff
@@ -1,22 +1,70 @@
-###############################################################################
-# Find QHULL
+#.rst:
+# FindQhull
+# --------
+#
+# Try to find QHULL library and headers. This module supports both old released versions
+# of QHULL ≤ 7.3.2 and newer development versions that ship with a modern config file.
+#
+# IMPORTED Targets
+# ^^^^^^^^^^^^^^^^
+#
+# This module defines the :prop_tgt:`IMPORTED` targets:
+#
+# ``QHULL::QHULL``
+#  Defined if the system has QHULL.
+#
+# Result Variables
+# ^^^^^^^^^^^^^^^^
+#
+# This module sets the following variables:
+#
+# ::
+#
+#   QHULL_FOUND               True in case QHULL is found, otherwise false
+#   QHULL_ROOT                Path to the root of found QHULL installation
+#
+# Example usage
+# ^^^^^^^^^^^^^
+#
+# ::
+#
+#     find_package(QHULL REQUIRED)
+#
+#     add_executable(foo foo.cc)
+#     target_link_libraries(foo QHULL::QHULL)
 #
-# This sets the following variables:
-# QHULL_FOUND - True if QHULL was found.
-# QHULL_INCLUDE_DIRS - Directories containing the QHULL include files.
-# QHULL_LIBRARIES - Libraries needed to use QHULL.
-# QHULL_DEFINITIONS - Compiler flags for QHULL.
-# If QHULL_USE_STATIC is specified then look for static libraries ONLY else
-# look for shared ones
+
+# Skip if QHULL::QHULL is already defined
+if(TARGET QHULL::QHULL)
+  return()
+endif()
+
+# Try to locate QHull using modern cmake config (available on latest Qhull version).
+find_package(Qhull NO_MODULE QUIET)
+if(Qhull_FOUND)
+  unset(Qhull_FOUND)
+  set(QHULL_FOUND ON)
+  add_library(QHULL::QHULL INTERFACE IMPORTED)
+  if(QHULL_USE_STATIC)
+    set_property(TARGET QHULL::QHULL APPEND PROPERTY INTERFACE_LINK_LIBRARY Qhull::qhullstatic)
+  else()
+    set_property(TARGET QHULL::QHULL APPEND PROPERTY INTERFACE_LINK_LIBRARY Qhull::libqhull)
+  endif()
+  get_filename_component(_config_dir "${Qhull_CONFIG}" DIRECTORY)
+  get_filename_component(QHULL_ROOT "${_config_dir}/../../.." ABSOLUTE)
+  return()
+endif()
 
 set(QHULL_MAJOR_VERSION 6)
 
 if(QHULL_USE_STATIC)
   set(QHULL_RELEASE_NAME qhullstatic)
   set(QHULL_DEBUG_NAME qhullstatic_d)
+  set(QHULL_LIBRARY_TYPE STATIC)
 else()
   set(QHULL_RELEASE_NAME qhull_p qhull${QHULL_MAJOR_VERSION} qhull)
   set(QHULL_DEBUG_NAME qhull_p_d qhull${QHULL_MAJOR_VERSION}_d qhull_d${QHULL_MAJOR_VERSION} qhull_d)
+  set(QHULL_LIBRARY_TYPE SHARED)
 endif()
 
 find_file(QHULL_HEADER
```

Comment:
This is still being executed even if the config file is found. Is it required? I would expect most this section to be entirely skipped if the config file is found.

Reply:
As I mentioned in the Discord this actually seems to be properly skipped when the config file is found.